File: output/xCodeEval_8000/C++/Java/a08055377482990fdb900d1f46aa4fc8.java
import java.util.*;

public class a08055377482990fdb900d1f46aa4fc8_nc {
    static final int MAGIC = 20000000;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        int[] p = new int[1000005];
        int[] to = new int[1000005];
        Pair[] pp = new Pair[1000005];
        Set<List<Integer>> s = new HashSet<>();
        List<List<Integer>> vv = new ArrayList<>();
        Queue<Pair<List<Integer>, Pair<Integer, Integer>>> q = new LinkedList<>();

        for (int i = 2; i < 1000005; i++) {
            if (p[i] == 0) {
                p[i] = i;
                if (i <= 1000005 / i) {
                    for (int j = i * i; j < 1000005; j += i) {
                        if (p[j] == 0) {
                            p[j] = i;
                        }
                    }
                }
            }
        }

        for (int i = 1; i <= 1000000; i++) {
            pp[i] = new Pair<>();
            pp[i].second = i;
            int la = -1;
            int ta = i;
            while (ta != 1) {
                if (p[ta] == la) {
                    pp[i].first.set(pp[i].first.size() - 1, pp[i].first.get(pp[i].first.size() - 1) + 1);
                } else {
                    pp[i].first.add(2);
                }
                la = p[ta];
                ta /= p[ta];
            }
            Collections.sort(pp[i].first, Collections.reverseOrder());
        }

        Arrays.sort(pp, 1, 1000001, new Comparator<Pair<List<Integer>, Integer>>() {
            @Override
            public int compare(Pair<List<Integer>, Integer> o1, Pair<List<Integer>, Integer> o2) {
                return o1.first.equals(o2.first) ? Integer.compare(o1.second, o2.second) : o1.first.compareTo(o2.first);
            }
        });

        for (int i = 1; i <= 1000000; i++) {
            if (i == 1 || !pp[i].first.equals(pp[i - 1].first)) {
                vv.add(pp[i].first);
            }
            to[pp[i].second] = vv.size() - 1;
        }

        for (int i = 0; i < vv.size(); i++) {
            for (int j = 0; j < i; j++) {
                x[i][j] = x[j][i] = xx[i][j] >= 'a' ? xx[i][j] - 'a' + 10 : xx[i][j] - '0';
            }
        }

        while (t-- > 0) {
            int ta = sc.nextInt();
            int tb = sc.nextInt();

            System.out.println(x[to[ta]][to[tb]]);
        }
    }
}

class Pair<T extends Comparable<T>, U extends Comparable<U>> implements Comparable<Pair<T, U>> {
    T first;
    U second;

    public Pair() {
    }

    public Pair(T first, U second) {
        this.first = first;
        this.second = second;
    }

    @Override
    public int compareTo(Pair<T, U> o) {
        return first.equals(o.first) ? second.compareTo(o.second) : first.compareTo(o.first);
    }
}

File: output/xCodeEval_remove_8000/C++/Go/a08055377482990fdb900d1f46aa4fc8.go
package main

import (
	"fmt"
	"sort"
)

var (
	t, la, ma, at int
	to            [1000005]int
	p             [1000005]int
	pp            [1000005]pair
	s             set
	m             [300]map[int]int
	x             [289][289]int
	xx            [289][299]byte
)

type pair struct {
	first  []int
	second int
}

type set map[vector]int

type vector []int

func go(aa vector, bb, cc int) {
	q := make([]pair, 0)
	q = append(q, pair{aa, pair{bb, cc}})
	s[aa] = true
	for len(q) > 0 {
		a := q[0].first
		b := q[0].second.first
		c := q[0].second.second
		q = q[1:]
		it := m[at][b]
		if it == 0 {
			m[at][b] = c
		} else {
			m[at][b] = min(it, c)
		}
		if c >= 10 {
			continue
		}
		c++
		for i := 0; i+1 < len(a); i++ {
			if a[i] != a[i+1] {
				a[i]--
				if _, ok := s[a]; !ok {
					s[a] = true
					q = append(q, pair{a, pair{b / (a[i] + 1) * a[i], c}})
				}
				a[i]++
			}
		}
		if len(a) > 0 {
			if a[len(a)-1] == 2 {
				a = a[:len(a)-1]
				if _, ok := s[a]; !ok {
					s[a] = true
					q = append(q, pair{a, pair{b / 2, c}})
				}
				a = append(a, 2)
			} else {
				a[len(a)-1]--
				if _, ok := s[a]; !ok {
					s[a] = true
					q = append(q, pair{a, pair{b / (a[len(a)-1] + 1) * a[len(a)-1], c}})
				}
				a[len(a)-1]++
			}
		}
		a = append(a, 2)
		if b*2 <= MAGIC {
			if _, ok := s[a]; !ok {
				s[a] = true
				q = append(q, pair{a, pair{b * 2, c}})
			}
		}
	}
	s = make(set)
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func main() {
	for i := 2; i < 1000005; i++ {
		if p[i] == 0 {
			p[i] = i
			if i <= 1000005/i {
				for j := i * i; j < 1000005; j += i {
					if p[j] == 0 {
						p[j] = i
					}
				}
			}
		}
	}

	for i := 1; i <= 1000000; i++ {
		pp[i].second = i
		la = -1
		ta = i
		for ta != 1 {
			if p[ta] == la {
				pp[i].first[len(pp[i].first)-1]++
			} else {
